src/java.desktop/share/classes/javax/swing/JTabbedPane.java

```diff
@@ -1789,9 +1790,7 @@ public String getToolTipText(MouseEvent event) {
     }
 
     private void checkIndex(int index) {
-        if (index < 0 || index >= pages.size()) {
-            throw new IndexOutOfBoundsException("Index: "+index+", Tab count: "+pages.size());
-        }
+        Objects.checkIndex(index, pages.size());
```

Comment:
I think we should not replace IOOBE checks with messages. Its better to keep it old way.

Reply:
Objects.checkIndex itself throws exceptions for wrong values. So this should be ok. No?

Reply:
We have specific message here like "Tab count" which we are not passing and will be helpful for debugging in future.

Reply:
Restored.